library/src/main/java/com/vinted/coper/CoperFragment.kt

```diff
@@ -101,7 +114,22 @@ internal class CoperFragment : Fragment() {
         return PermissionResult.Denied(deniedPermissions)
     }
 
-    private fun requestPermissionAsync(
+    // On devices with lower sdk than 23, there is no requesting permissions.
+    private suspend fun requestPermissionsByVersion(
+        permissions: Array<out String>
+    ): List<PermissionCheckResult> {
+        return if (isDeviceSdkAtLeast23()) {
+            requestPermissionsAsync(permissions).await()
+        } else {
+            permissions.map { PermissionCheckResult.Denied(it) }
```

Comment:
I suppose you should report permission granted in this situation. But before I explaining why maybe you should just drop support of <=23? For example, you could allow request permission only on `>=23`. Otherwise, you should change it to granted, because, let imagine we have a code like:
```
coper.request(CAMERA)).onSuccess {
  openCamera()
}
```
If the user runs <23 API version, this code do not work even if we already give CAMERA permissions via manifest. To fix it, we have to additional `if` to check version. However, very likely you will not be aware of such a situation, and probably you newer test it, but your users do.